### DISCUSSION TITLE
Are Yup validations still 100% supported?

### DISCUSSION THREAD

**Discussion by Cuantico** [react-hook-form/react-hook-form#5916] [General]
Hi, I updated to version 7 and I immediately got this error:
```
Uncaught (in promise) TypeError: Cannot read property 'reduce' of undefined
    at yup.ts:14
    at async useForm.ts:996
```

**Details:**
-update react-hook-form from 6.15.4 to 7.10.1
-update @hookform/resolvers from 1.3.5 to 2.6.0

Are Yup validations still 100% supported?

**Comment by bluebill1049** [comment]
yes, provide a CSB, we can take a look at it.

**Comment by Cuantico** [comment]
https://codesandbox.io/s/provider-perf-forked-2n5op?file=/src/App.js

this code doesn't break with react-hook-form from 6.15.4 & @hookform/resolvers from 1.3.5 but it does with  react-hook-form from 7.10.1 &  @hookform/resolvers from 2.6.0, you can switch the versions to check it out.
```
test2: yup.object().shape({
    test2: yup.string().when("test", {
      is: (value) => value.length > 0,
      then: yup.string().required("Errorrrrrrrrr")
    })
```
    A refactor in the Yup Schema solves the Issue I think:
  ```
 test2: yup.object().shape({
    test2: yup.string().when("test", {
      is: (value) => value.?length > 0,
      then: yup.string().required("Errorrrrrrrrr")
    })
```
